modules/c++/mem/tests/ScratchVisualization.cpp

```diff
@@ -0,0 +1,477 @@
+#include <mem/ScratchMemory.h>
+#include <mem/BufferView.h>
+#include <cli/ArgumentParser.h>
+#include <str/Convert.h>
+#include <vector>
+#include <fstream>
+#include <string>
+
+struct Operation
+{
+    Operation(const std::string op,
+              const std::string name,
+              const size_t bytes) :
+        op(op),
+        name(name),
+        bytes(bytes)
+    {
+    }
+
+    std::string op;
+    std::string name;
+    size_t bytes;
+    mem::BufferView<sys::ubyte> buffer;
+};
+
+/*!
+ * \class Visualizer
+ *
+ * \brief Handles the inclusion and visualization of a put or release operation
+ */
+class Visualizer
+{
+public:
+    /*!
+     * Constructor
+     *
+     * \param prevOperations All operations handled in previous iterations
+     * \param iteration Which iteration is being processed
+     * \param[in,out] htmlFile Ofstream to .html file
+     * \param[in,out] cssFile Ofstream to .css file
+     */
+    Visualizer(const std::vector<Operation> prevOperations,
+               const size_t iteration,
+               std::ofstream& htmlFile,
+               std::ofstream& cssFile):
+        mPrevOperations(prevOperations),
+        mIteration(iteration),
+        mHTMLFile(htmlFile),
+        mCSSFile(cssFile)
+    {
+        mStartPtr = NULL;
+        mColors.push_back("lightgrey");
+        mColors.push_back("lightblue");
+        mColors.push_back("lightcyan");
+        mReleasedColors.push_back("indianred");
+        mReleasedColors.push_back("coral");
+        mReleasedColors.push_back("salmon");
+    }
+
+    /*!
+     * Find the lowest start address, which dictates the first block, and set mStartPtr.
+     *
+     * \param operations A vector of all operations
+     */
+    void set_start_ptr(const std::vector<Operation>& operations)
+    {
+        mStartPtr = operations.at(0).buffer.data;
+        for (size_t ii = 0; ii < operations.size(); ++ii)
+        {
+            if (operations.at(ii).buffer.data - mStartPtr <= 0)
+            {
+                mStartPtr = operations.at(ii).buffer.data;
+            }
+        }
+    }
+
+    /*!
+     * Write the CSS and HTML to create a box for a single operation.
+     *
+     * \param op Operation to draw a box of
+     * \param color_iter Iterator to arbitrarily select a color
+     */
+    void create_box(const Operation op, const size_t color_iter)
```

Comment:
Rename this to createBox. Pass op as const ref and remove const from color_iter. Rename color_iter to colorIter.